lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm

```diff
@@ -95,11 +95,18 @@ sub list {
 
 # check existing job group on top level to prevent create/update duplicate
 sub check_top_level_group {
-    my ($self) = @_;
+    my ($self, $group_id) = @_;
 
     return 0 if $self->is_parent;
     my $properties = $self->load_properties;
-    return $self->resultset->search({name => $properties->{name}, parent_id => undef});
+    my $conditions = {name => $properties->{name}, parent_id => undef};
+    unless ($group_id) {
+        return $self->resultset->search($conditions);
```

Comment:
now this line is the same as 3 lines later. So please just move the $conditions->{id} = into an if - and return the search uncondiotionally

Reply:
yes, got it...